### PR TITLE
Add ImageSeriesReader SpacingWarningRelThreshold accessors

### LINKED ISSUE #2034: Add ImageSeriesReader::SpacingWarningRelThreshold
This additional option was introduced in ITK here:
https://github.com/InsightSoftwareConsortium/ITK/pull/1616

And the warning has been discussed here:
https://github.com/InsightSoftwareConsortium/ITK/issues/1607

### PR DESCRIPTION
**Pull request by blowekamp**
Closes #2034.

### CODE REVIEW

**Review by dave3d — APPROVED** [review]

**Review by zivy — APPROVED** [review]
Minor comment about settings harmonization (not sure about it).

File: Code/IO/include/sitkImageSeriesReader.h [hunk 1]
```diff
@@ -234,12 +240,14 @@ class SITKIO_EXPORT ImageSeriesReader : public ImageReaderBase
...
 
   // Holder of process object for active measurements
-  itk::ProcessObject * m_Filter;
+  itk::ProcessObject * m_Filter{nullptr};
 
 
   std::vector<PathType> m_FileNames;
 
-  bool m_MetaDataDictionaryArrayUpdate;
+  double m_SpacingWarningRelThreshold{1e-4};
```
[the remark this answers is not in the document]
Comment (blowekamp) [reply Code/IO/include/sitkImageSeriesReader.h:248]: It's not clear what you are referring too, this value matches the value in ITK. I didn't write a test to ensure the defaults match.

[the remark this answers is not in the document]
Comment (zivy) [reply Code/IO/include/sitkImageSeriesReader.h:248]: I don't remember the name of the ITK method, but I believe there is one which compares equality-up-to-epsilon of image metadata. That function has an epsilon value for equivalence of two spacings. I was wondering if the same epsilon value should be used for this spacing comparison too.

[the remark this answers is not in the document]
Comment (blowekamp) [reply Code/IO/include/sitkImageSeriesReader.h:248]: Are you suggesting that we should deviate from the values defined in ITK? You can find the original ITK discussion here: https://github.com/InsightSoftwareConsortium/ITK/issues/1607

[the remark this answers is not in the document]
Comment (zivy) [reply Code/IO/include/sitkImageSeriesReader.h:248]: No. Not suggesting we deviate from the ITK value. I thought ITK was using the same value for this comparison and the comparison done to ensure that two images occupy the same physical space prior to performing an operation (e.g. AddImageFilter). I see ITK has a dedicated value for the series reader which is repeated here, so consistent with the ITK inconsistency. 

Bottom line, this is fine, just a bit inconsistent in ITK.

File: Testing/Unit/sitkImageIOTests.cxx [hunk 2]
```diff
@@ -514,6 +515,68 @@ TEST(IO, DicomSeriesReader)
...
+    if (i == image.GetSize()[2] - 1)
+    {
+      origin[2] += 1e-5;
+      slice.SetOrigin(origin);
+    }
+
+    sitk::WriteImage(slice, seriesPaths[i].string());
+  };
+
+  reader.SetFileNames(std::vector<std::string>{ seriesPaths.begin(), seriesPaths.end() });
```
Comment (dave3d) [inline Testing/Unit/sitkImageIOTests.cxx:554]: This is the probem line on Windows

File: Testing/Unit/sitkImageIOTests.cxx [hunk 3]
```diff
@@ -514,6 +515,68 @@ TEST(IO, DicomSeriesReader)
...
+    std::filesystem::remove_all(seriesDir);
+  }
+  std::filesystem::create_directories(seriesDir);
+
+  const std::vector<std::filesystem::path> seriesPaths{
+    seriesDir / "000000.mha",
+    seriesDir / "000001.mha",
+    seriesDir / "000002.mha"
+  };
+  auto image = sitk::Image(10, 10, seriesPaths.size(), sitk::sitkUInt8);
```
Comment (dave3d) [inline Testing/Unit/sitkImageIOTests.cxx:536]: Cast 3rd arg to (unsigned int)

**Pull request merged**